lib/HLSL/DxilPromoteResourcePasses.cpp

```diff
@@ -219,7 +220,8 @@ bool DxilPromoteStaticResources::PromoteStaticGlobalResources(
       Insts.clear();
     }
     if (!bUpdated) {
-      M.getContext().emitError(dxilutil::kResourceMapErrorMsg);
+      llvm_unreachable(
+          "local resource not guaranteed to map to unique global resource.");
```

Comment:
I'd like to understand better how these were identified as unreachable. They are in a pass that comes before the only untouched reporting of this error in LowerCreateHandleForLib. This is a rather squirrelly error that often only fires with no optimizations enabled. I worry that these might be triggerable here after all.